boards/arm/dragonclaw/Kconfig.defconfig

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022 Google Inc
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_DRAGONCLAW
+
+config BOARD
+	default "dragonclaw"
```

Comment:
Should we get into the habit of prefixing codenamed boards with `google_`? For example we currently have `google_kukui` already, would make it easy to spot them and give us a change to have a common maintainer or codeowner entry if needed. @keith-zephyr for an extra opinion

Reply:
Let's wait for Keith's opinion.

Reply:
I agree with Fabio. Let's prefix boards with `google_`.

Reply:
Done! Thanks!